src/server/project.ts

```diff
@@ -1188,6 +1145,34 @@ namespace ts.server {
         }
     }
 
+    function getUnresolvedImports(program: Program, cachedUnresolvedImportsPerFile: Map<ReadonlyArray<string>>): SortedReadonlyArray<string> {
+        const ambientModules = program.getTypeChecker().getAmbientModules().map(mod => stripQuotes(mod.getName()));
+        return toDeduplicatedSortedArray(flatMap(program.getSourceFiles(), sourceFile =>
+            extractUnresolvedImportsFromSourceFile(sourceFile, ambientModules, cachedUnresolvedImportsPerFile)));
+    }
+
+    function extractUnresolvedImportsFromSourceFile(file: SourceFile, ambientModules: ReadonlyArray<string>, cachedUnresolvedImportsPerFile: Map<ReadonlyArray<string>>): ReadonlyArray<string> {
+        const cached = cachedUnresolvedImportsPerFile.get(file.path);
+        if (cached) {
+            // found cached result, return
+            return cached;
+        }
+        let unresolvedImports: string[] | undefined;
+        if (file.resolvedModules) {
+            file.resolvedModules.forEach((resolvedModule, name) => {
+                // pick unresolved non-relative names
+                if ((!resolvedModule || !extensionIsTS(resolvedModule.extension)) &&

```

Comment:
You should also ignore json resolution ?

Reply:
I don't think this would come up, because this code will only apply to non-relative imports, and we only resolve to `.json` for relative imports with an explicit `.json` extension, right?

Reply:
no we resolve any json resolution with ".json" extension

Reply:
it will resolve to "node_modules/foo/index.json" when you import "foo.json" or import of "foo/lib.json" will resolve to "node_modules/foo/lib.json"?

Reply:
Ah OK, "foo/lib.json" can import a json module. Shouldn't we still look for a type definition for the package though?